### PR TITLE
Add scene change breadcrumb to Unity

### PR DESCRIPTION
**Pull request by fractalwrench**
Records when scenes load, and logs a Breadcrumb with the scene name.

### CODE REVIEW

File: example/Assets/Standard Assets/Bugsnag/Editor/BugsnagPostProcess.cs [hunk 1]
```diff
@@ -47,12 +47,6 @@ public static void OnPostProcessBuild(BuildTarget target, string path)
         var sbWeb = new StringBuilder ();
         foreach (var line in indexLines) {
             sbWeb.AppendLine (line);
-
-            // Add an extra line below the first script tag, linking to the js notifier
```
Comment (snmaynard) [inline example/Assets/Standard Assets/Bugsnag/Editor/BugsnagPostProcess.cs]: Why has this been removed as a part of these changes?

File: example/Assets/Standard Assets/Bugsnag/Bugsnag.cs [hunk 2]
```diff
@@ -72,7 +73,12 @@ public class NativeBugsnag {
             AndroidJavaObject activity = unityPlayerClass.GetStatic<AndroidJavaObject>("currentActivity");
             AndroidJavaObject app = activity.Call<AndroidJavaObject>("getApplicationContext");
 
-            Bugsnag.CallStatic<AndroidJavaObject> ("init", app, apiKey);
+            jvalue[] args =  new jvalue[2] {
+                new jvalue() { l = app.GetRawObject() },
+                new jvalue() { l = AndroidJNI.NewStringUTF(apiKey) },
+            };
+            IntPtr methodId = AndroidJNI.GetStaticMethodID(BugsnagUnity.GetRawClass(), "init", "(Landroid/content/Context;Ljava/lang/String;)V");
+            AndroidJNI.CallStaticVoidMethod(BugsnagUnity.GetRawClass(), methodId, args);
```
Comment (snmaynard) [inline example/Assets/Standard Assets/Bugsnag/Bugsnag.cs]: Why are we using the JNI directly rather than going through the unity wrapper for it?

File: example/Assets/Standard Assets/Bugsnag/Bugsnag.cs [hunk 3]
```diff
@@ -529,7 +521,8 @@ public enum Severity {
                 errorClass = match.Groups["errorClass"].Value;
                 errorMessage = match.Groups["message"].Value.Trim();
             } else {
-                errorClass = logString;
+                errorClass = "UnityLog" + type;
```
Comment (snmaynard) [inline example/Assets/Standard Assets/Bugsnag/Bugsnag.cs]: This is for an unrelated change? This seems to already be on master. What is going on here?

File: example/Assets/Standard Assets/Bugsnag/Editor/BugsnagPostProcess.cs [hunk 4]
```diff
@@ -120,7 +114,7 @@ public static void OnPostProcessBuild(BuildTarget target, string path)
...
                     "\t\t\tshellPath = \"/usr/bin/env ruby\";\n" +
-                    "\t\t\tshellScript = \"# bugsnag dsym upload script\\nfork do\\n  Process.setsid\\n  STDIN.reopen(\\\"/dev/null\\\")\\n  STDOUT.reopen(\\\"/dev/null\\\", \\\"a\\\")\\n  STDERR.reopen(\\\"/dev/null\\\", \\\"a\\\")\\n\\n  require 'shellwords'\\n\\n  Dir[\\\"#{ENV[\\\"DWARF_DSYM_FOLDER_PATH\\\"]}/*/Contents/Resources/DWARF/*\\\"].each do |dsym|\\n    system(\\\"curl -F dsym=@#{Shellwords.escape(dsym)} -F projectRoot=#{Shellwords.escape(ENV[\\\"PROJECT_DIR\\\"])} https://upload.bugsnag.com/\\\")\\n  end\\nend\";\n" +
+                    "\t\t\tshellScript = \"# bugsnag dsym upload script\\nfork do\\n  Process.setsid\\n  STDIN.reopen(\\\"/dev/null\\\")\\n  STDOUT.reopen(\\\"/dev/null\\\", \\\"a\\\")\\n  STDERR.reopen(\\\"/dev/null\\\", \\\"a\\\")\\n\\n  require \\\"shellwords\\\"\\n\\n  Dir[\\\"#{ENV[\\\"DWARF_DSYM_FOLDER_PATH\\\"]}/*/Contents/Resources/DWARF/*\\\"].each do |dsym|\\n    system(\\\"curl -F dsym=@#{Shellwords.escape(dsym)} -F projectRoot=#{Shellwords.escape(ENV[\\\"PROJECT_DIR\\\"])} https://upload.bugsnag.com/\\\")\\n  end\\nend\";\n" +
```
Comment (snmaynard) [inline example/Assets/Standard Assets/Bugsnag/Editor/BugsnagPostProcess.cs]: Why have we changed quotes here? Is this an unrelated change? Possibly already on master?

**Comment by fractalwrench** [comment]
@snmaynard the only change which I've made to the code in the notifier itself is [here](https://github.com/bugsnag/bugsnag-unity/pull/54/files#diff-bf5567ac1f5718f2297247e57f0555dd)

I've been following the Contributing.MD instructions, which runs `rake update` and copies the C# over to the example folder. I'm assuming that this wasn't done the last time we merged to master?

**Comment by snmaynard** [comment]
Sounds likely. Perhaps we can run that against master now and then the diff will be more manageable.

It looks like the diff will just be the main Bugsnag.cs file which is like 2 lines.

**Comment by fractalwrench** [comment]
@snmaynard wouldn't we want to keep the example code up to date with the latest plugin code? I've run `rake update` on master locally, and confirmed that it updates those two files in the example folder.

**Comment by snmaynard** [comment]
Yeah `rake update` should always be a no op on master

**Comment by snmaynard** [comment]
I was just saying if you run `rake update` on master then merge master into this branch, the diff here should be tiny. From what I can see this pr looks fine though.

File: src/Assets/Standard Assets/Bugsnag/Bugsnag.cs [hunk 5]
```diff
@@ -450,10 +450,12 @@ public enum Severity {
 #if UNITY_5_4_OR_NEWER
     void SceneLoaded(Scene scene, LoadSceneMode mode) {
         Bugsnag.Context = scene.name;
+		LeaveBreadcrumb("Scene loaded: " + scene.name);
     }
 #else
     void OnLevelWasLoaded(int level) {
         Bugsnag.Context = GetLevelName();
+		LeaveBreadcrumb("Scene loaded: " + scene.name);
```
Comment (snmaynard) [inline src/Assets/Standard Assets/Bugsnag/Bugsnag.cs]: scene wont be defined here?

**Comment by fractalwrench** [comment]
@snmaynard I've merged the rake update into master and the diff for this PR is now indeed tiny

**Comment by snmaynard** [comment]
Nice one @fractalwrench - all makes sense now!
